src/invenio-files-js/controllers/InvenioFilesCtrl.js

```diff
@@ -234,6 +234,27 @@ function InvenioFilesCtrl($rootScope, $scope, $q, $timeout,
         Uploader.pushToQueue(file);
       }
     });
+    if(vm.invenioFilesArgs.autoStartUpload && vm.hasQueuedUploads()){
+      vm.upload()
+    }
+  }
+
+  /**
+  * Check if there are files queued for uploading and there is no active 
+  * upload
+  * @memberof InvenioFilesCtrl
+  * @function hasQueuedUploads
+  */
+  function hasQueuedUploads() {
+    runUpload = false
```

Comment:
```suggestion
    result = false
```

Naming this `runUpload` is like jumping the gun about what's going to happen after this function is called